[3.4.0] Compilation failure on 32 bit platforms

On Debian's 32-bit platforms (i386, x32, ARM, ppc, hppa), the compilation fails with the following error:
```
cd /<<PKGBUILDDIR>>/obj-i686-linux-gnu/tables && /usr/bin/c++ -DCFITSIO_VERSION_MAJOR=4 -DCFITSIO_VERSION_MINOR=0 -DHAVE_FFTW3 -DHAVE_FFTW3_THREADS -DHAVE_O_DIRECT -DHAVE_READLINE -DUSE_THREADS -DWCSLIB_VERSION_MAJOR=7 -DWCSLIB_VERSION_MINOR=7 -Dcasa_tables_EXPORTS -I/<<PKGBUILDDIR>> -I/<<PKGBUILDDIR>>/obj-i686-linux-gnu -I/<<PKGBUILDDIR>>/obj-i686-linux-gnu/tables -fcx-fortran-rules -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -ffloat-store -Wdate-time -D_FORTIFY_SOURCE=2 -fsigned-char -std=c++11 -pthread -fPIC -MD -MT tables/CMakeFiles/casa_tables.dir/Tables/RowCopier.cc.o -MF CMakeFiles/casa_tables.dir/Tables/RowCopier.cc.o.d -o CMakeFiles/casa_tables.dir/Tables/RowCopier.cc.o -c /<<PKGBUILDDIR>>/tables/Tables/RowCopier.cc
/<<PKGBUILDDIR>>/tables/Tables/RefTable.cc: In constructor ‘casacore::RefTable::RefTable(casacore::BaseTable*, const casacore::Vector<bool, std::allocator<bool> >&)’:
/<<PKGBUILDDIR>>/tables/Tables/RefTable.cc:118:52: error: call of overloaded ‘min(size_t, casacore::rownr_t)’ is ambiguous
  118 |     rownr_t nr = min (mask.nelements(), btp->nrow());
      |                                                    ^
In file included from /<<PKGBUILDDIR>>/casacore/casa/Utilities/DataType.h:33,
                 from /<<PKGBUILDDIR>>/casacore/tables/Tables/BaseColDesc.h:34,
                 from /<<PKGBUILDDIR>>/casacore/tables/Tables/ColumnDesc.h:34,
                 from /<<PKGBUILDDIR>>/casacore/tables/Tables/ColDescSet.h:33,
                 from /<<PKGBUILDDIR>>/casacore/tables/Tables/TableDesc.h:34,
                 from /<<PKGBUILDDIR>>/casacore/tables/Tables/BaseTable.h:35,
                 from /<<PKGBUILDDIR>>/casacore/tables/Tables/RefTable.h:34,
                 from /<<PKGBUILDDIR>>/tables/Tables/RefTable.cc:28:
/<<PKGBUILDDIR>>/casacore/casa/BasicSL/Complex.h:295:16: note: candidate: ‘casacore::Complex casacore::min(const Complex&, const Complex&)’
  295 | inline Complex min(const Complex &x, const Complex &y)
      |                ^~~
/<<PKGBUILDDIR>>/casacore/casa/BasicSL/Complex.h:297:17: note: candidate: ‘casacore::DComplex casacore::min(const DComplex&, const DComplex&)’
  297 | inline DComplex min(const DComplex &x, const DComplex &y)
      |                 ^~~
In file included from /<<PKGBUILDDIR>>/tables/Tables/RefTable.cc:40:
/<<PKGBUILDDIR>>/casacore/casa/BasicMath/Math.h:159:12: note: candidate: ‘casacore::Int casacore::min(casacore::Int, casacore::Int)’
  159 | inline Int min(Int a, Int b) { if (a > b) return b; else return a; }
      |            ^~~
/<<PKGBUILDDIR>>/casacore/casa/BasicMath/Math.h:162:13: note: candidate: ‘casacore::uInt casacore::min(casacore::uInt, casacore::uInt)’
  162 | inline uInt min(uInt a, uInt b){ if (a>b) return b; else return a; }
      |             ^~~
/<<PKGBUILDDIR>>/casacore/casa/BasicMath/Math.h:165:15: note: candidate: ‘casacore::uInt64 casacore::min(casacore::uInt64, casacore::uInt64)’
  165 | inline uInt64 min(uInt64 a, uInt64 b){ if (a>b) return b; else return a; }
      |               ^~~
/<<PKGBUILDDIR>>/casacore/casa/BasicMath/Math.h:168:15: note: candidate: ‘casacore::Double casacore::min(casacore::Double, casacore::Double)’
  168 | inline Double min(Double a, Double b) { if (a > b) return b; else return a; }
      |               ^~~
/<<PKGBUILDDIR>>/casacore/casa/BasicMath/Math.h:170:15: note: candidate: ‘casacore::Double casacore::min(casacore::Double, casacore::Float)’
  170 | inline Double min(Double a, Float b) { if (a > b) return b; else return a; }
      |               ^~~
/<<PKGBUILDDIR>>/casacore/casa/BasicMath/Math.h:172:15: note: candidate: ‘casacore::Double casacore::min(casacore::Float, casacore::Double)’
  172 | inline Double min(Float a, Double b) { if (a > b) return b; else return a; }
      |               ^~~
/<<PKGBUILDDIR>>/casacore/casa/BasicMath/Math.h:175:14: note: candidate: ‘casacore::Float casacore::min(casacore::Float, casacore::Float)’
  175 | inline Float min(Float a, Float b) { if (a > b) return b; else return a; }
      |              ^~~
```
Full build log for i386 [here](https://buildd.debian.org/status/fetch.php?pkg=casacore&arch=i386&ver=3.4.0-1&stamp=1632099987&raw=0).

--- a/tables/Tables/RefTable.cc
+++ b/tables/Tables/RefTable.cc
@@ -115,7 +115,7 @@ RefTable::RefTable (BaseTable* btp, const Vector<Bool>& mask)
     tdescPtr_p = new TableDesc (btp->tableDesc(), TableDesc::Scratch);
     setup (btp, Vector<String>());
     //# Store the rownr if the mask is set.
-    rownr_t nr = min (mask.nelements(), btp->nrow());
+    rownr_t nr = std::min<rownr_t> (mask.nelements(), btp->nrow());
     for (rownr_t i=0; i<nr; i++) {
 	if (mask(i)) {
 	    addRownr (i);

